### PR TITLE
Do not change Win32 lasterror in the first MONO_EXIT_GC_SAFE and remove unused W32_UPDATE_LAST_ERROR_RESTORE_POINT.

### PR DESCRIPTION
**Pull request by jaykrell**

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/utils/mono-threads-coop.c [hunk 1]
```diff
@@ -648,10 +648,16 @@ mono_threads_suspend_policy (void)
...
 		// otherwise use full preemptive suspend.
+
+		W32_DEFINE_LAST_ERROR_RESTORE_POINT;
+
 		   (policy = threads_suspend_policy_getenv ())
 		|| (policy = threads_suspend_policy_default ())
 		|| (policy = threads_suspend_policy_getenv_compat ())
 		|| (policy = MONO_THREADS_SUSPEND_FULL_PREEMPTIVE);
+
+		W32_UPDATE_LAST_ERROR_RESTORE_POINT;
```
Comment (lambdageek) [inline mono/utils/mono-threads-coop.c]: This should be `W32_RESTORE_LAST_ERROR_FROM_RESTORE_POINT`, I think?

Comment (jaykrell) [reply mono/utils/mono-threads-coop.c]: Oops probably. I thought there’d only be two symbols, found these two, glossed over the definition, thought the name sounded funny, couldn’t imagine a third symbol. Will look again.

Comment (jaykrell) [reply mono/utils/mono-threads-coop.c]: Fixed. I also suggest removing `W32_UPDATE_LAST_ERROR_RESTORE_POINT`.
It isn't used. I have preserved lasterror a fair amount through the years, using a local with constructor/destructor, and don't recall a use-case for this extra piece.

**Comment by jaykrell** [comment]
@lateralusX Use-case for `W32_UPDATE_LAST_ERROR_RESTORE_POINT`?

**Comment by jaykrell** [comment]
@monojenkins build failed

**Pull request merged**